tests/test.js

```diff
@@ -1,50 +1,120 @@
-  it('decrytion of encrypted value should be same as original value', () =>{
+describe('crypter test', function () {
+  it('decrytion of encrypted value should be same as original value', function () {
     let demoValue = 'testData'
-    assert.equal(demoValue, crypter.decrypt(crypter.encrypt(demoValue)))
-  })
-})
-
-describe('WebUi Generator', () => {
-  let uniqueId = uuidV4()
-  let email = 'fossasia@gmail.com'
-  let args = [
-    "-g", "https://github.com/fossasia/yaydoc.git",
-    "-t", "alabaster",
-    "-m", email,
-    "-u", uniqueId,
-    "-w", "true"
-  ]
-  let exitCode
-  let files
+    assert.equal(demoValue, crypter.decrypt(crypter.encrypt(demoValue)));
+  });
+});
 
-  before((done) => {
-    let process = spawn('./generate.sh', args)
-    process.on('exit', (code) => {
-      exitCode = code
-      files = fs.readdirSync(`temp/${email}`)
-      done()
+sinon.stub(require("child_process"), "spawn").callsFake(function (fileName, args) {
+  if (fileName !== "./ghpages_deploy.sh" && fileName !== "./heroku_deploy.sh" && fileName !== "./generate.sh") {
+    throw new Error(`invalid ${fileName} invoked`);
+  }
+  if (fileName === "./ghpages_deploy.sh") {
+    let ghArgs = ["-e", "-i", "-w", "-n", "-o", "-r"];
+    ghArgs.forEach(function (x) {
+      if (args.indexOf(x) < 0) {
+        throw new Error(`${x} argument is not passed`);
+      }
     })
-  })
-  it('exit code should be zero', () => {
-    assert.equal(exitCode, 0)
-  })
-  it('preview folder should be created', () => {
-    if(files.indexOf(`${uniqueId}_preview`) < 0){
-      throw new Error('preview not created')
+  }
+  if (fileName === "./heroku_deploy.sh") {
+    let herokuArgs = ["-e", "-u", "-w", "-h", "-n"];
+    herokuArgs.forEach(function (x) {
+      if (args.indexOf(x) < 0) {
+        throw new Error(`${x} argument is not passed`);
+      }
+    });
+  }
+  if (fileName === "./generate.sh") {
+    let generateArgs = ["-g", "-t", "-m", "-d", "-u", "-w", "-s", "-p"];
+    generateArgs.forEach(function (x) {
+      if (args.indexOf(x) < 0) {
+        throw new Error(`${x} argument is not passed`);
+      }
+    });
+  }
+  let process = {
+    on: function (listenerId, callback) {
+      if (listenerId !== "exit") {
+        throw new Error("listener id is not exit");
+      }
     }
-  })
-  it('zipped documentation should be created', () => {
-    if(files.indexOf(`${uniqueId}.zip`) < 0){
-      throw new Error('zipped documentation not created')
-    }
-  })
-})
+  }
+  return process;
+});
+function lineOutput(socket, process, event, increment) {
+  if (socket === undefined) {
+    throw Error("socket object is not passed in the lineOutput");
+  }
+  if (process === undefined) {
+    throw Error("process object is not passed in the lineOutput");
+  }
+  if (event === undefined) {
+    throw Error("event object is not passed in the lineOutput");
+  }
+  if (increment === undefined) {
+    throw Error("increment object is not passed in the lineOutput");
+  }
```

Comment:
- increment has to be an integer value
- increment cannot be <= 0 
- increment cannot be > 100